sci-biology/salmon/salmon-1.4.0.ebuild

```diff
@@ -0,0 +1,27 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake
+
+DESCRIPTION="Transcript-level quantification from RNA-seq reads using lightweight alignments"
+HOMEPAGE="https://github.com/COMBINE-lab/salmon"
+SRC_URI="https://github.com/COMBINE-lab/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	https://github.com/COMBINE-lab/pufferfish/archive/${PN}-v${PV}.tar.gz -> ${P}-pufferfish.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+BDEPEND="net-misc/curl
+		app-arch/unzip
+		sys-libs/zlib
+		app-arch/bzip2
+		app-arch/xz-utils
+		dev-libs/boost:0[threads]
```

Comment:
boost in BDEPEND, is this true?
And the dep is IMO missing an `=` after the slot
Although I don't think, this will solve the issue

Reply:
> boost in BDEPEND, is this true?

I think so, at least it was there before I started changing things, and it does seem to want it at build time (the configure phase fails with the error that it is missing, so it is definitely looking for boost)

> And the dep is IMO missing an = after the slot

Boost only has one slot (so I guess the `:0` is useless and should go?)

Reply:
> > boost in BDEPEND, is this true?
> 
> I think so, at least it was there before I started changing things, and it does seem to want it at build time (the configure phase fails with the error that it is missing, so it is definitely looking for boost)

But then it uses the dependencies in a wrong way. BDEPEND should only contain packages on the build machine (CBUILD), which need not be available on the machine the package is going to get installed (CHOST). Just using RDEPEND=${BDEPEND} is, in my opinion no clean way. I doubt you would get such a construction merged on ::gentoo.
Have a package available at build time is reached by using DEPEND=${RDEPEND}.
https://devmanual.gentoo.org/general-concepts/dependencies/index.html

> 
> > And the dep is IMO missing an = after the slot
> 
> Boost only has one slot (so I guess the `:0` is useless and should go?)

The 0 can go away, but it needs a `:=` slot operator, as it has subslots, to force a rebuild after boost get's updated.

Reply:
Like this?

Reply:
Yes this looks better. But you have to check the dependencies carefully. zlib is most likely a runtime dependency. For tbb, it depends how it's used. If the library is linked, it's RDEPEND, if only the template headers are used and no linking is done, it can probably be BDEPEND. For curl, is the fetchPufferfish.sh script installed or only used during build time? For the first, curl needs to be in RDEPEND, for the latter BDEPEND would IMO be sufficient. Just check whether the dependencies are needed on CHOST (RDEPEND) or **only** on CBUILD (BDEPEND).

Reply:
It probably doesn't have to depend on curl or unzip at all, because the patch removes those commands from the fetchPutterfish.sh script

Reply:
> It probably doesn't have to depend on curl or unzip at all, because the patch removes those commands from the fetchPutterfish.sh script

Nope, I was wrong